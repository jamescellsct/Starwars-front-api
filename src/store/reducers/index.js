import { combineReducers } from "redux";

// import dos reducers
import pagination from "./pagination";
import peoples from "./peoples";
import planets from "./planets";
import starships from "./starships";
import vehicles from "./vehicles";

// função que retorna um 'container' de reducers
export default combineReducers({
  pagination,
  peoples,
  planets,
  starships,
  vehicles
});
