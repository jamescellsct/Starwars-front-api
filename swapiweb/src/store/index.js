import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";

// importando o 'container' de reducers como rootReducer
import rootReducer from "./reducers";

/*
    CRIANDO O ESTADO GLOBAL DA APLICAÇÃO   

    Essa função createStore() recebe como parâmetro o stado da aplicação,
    podendo ser qualquer coisa.
    Nesse caso foi passado o 'container' de reducers, onde contém as
    funções e lógicas que retornam um estado.

*/
const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
