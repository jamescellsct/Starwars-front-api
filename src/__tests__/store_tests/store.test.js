import store from "../../store";

describe("Testing store redux component", () => {
  it("Should render correctly store with initial state", () => {
    const expectedState = {
      pagination: { page: 1 },
      peoples: null,
      planets: null,
      starships: null,
      vehicles: null
    };
    expect(store.getState()).toEqual(expectedState);
    expect(store.getState()).not.toBe(null);
    expect(store.getState()).toMatchSnapshot();
  });
});

console.log(store);
