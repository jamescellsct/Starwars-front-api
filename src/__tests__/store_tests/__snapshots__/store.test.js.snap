// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing store redux component Should render correctly store with initial state 1`] = `
Object {
  "pagination": Object {
    "page": 1,
  },
  "peoples": null,
  "planets": null,
  "starships": null,
  "vehicles": null,
}
`;
